### PR TITLE
Dlq cursor reset

### PR DESCRIPTION
**Pull request by fghibellini**
# One-line summary

> Zalando ticket : ARUHA-1531

## Description
A few sentences describing the overall goals of the pull request's
commits.

## Review
- [ ] Tests
- [ ] Documentation

## Deployment Notes
These should highlight any db migrations, feature toggles, etc.

### CODE REVIEW

File: acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java [hunk 1]
```diff
@@ -862,4 +880,28 @@ private Subscription createAutoDLQSubscription(final EventType eventType,
                         SUBSCRIPTION_UNPROCESSABLE_EVENT_POLICY, unprocessableEventPolicy.toString()));
         return createSubscription(subscription);
     }
+
+    List<SubscriptionCursor> getSubscriptionCursors(String sid) throws IOException {
+        var response = given()
```
Comment (a1exsh) [inline acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java:885]: Let's avoid `var`s.

File: api-cursors/src/main/java/org/zalando/nakadi/service/CursorsService.java [hunk 2]
```diff
@@ -223,8 +225,19 @@ public void resetCursors(final String subscriptionId,
...
 
-                zkClient.closeSubscriptionStreams(() -> zkClient.forceCommitOffsets(newCursors), timeout);
+                zkClient.closeSubscriptionStreams(() -> {
+                    zkClient.forceCommitOffsets(newCursors);
+                    // reset the DLQ state of any partition that got updated
+                    zkClient.updateTopology(topology ->
+                        Arrays
+                            .stream(topology.getPartitions())
+                            .filter(isUpdatedPartition)
+                            .map(p -> p.toZeroFailedCommits().toLastDeadLetterOffset(null))
```
Comment (a1exsh) [inline api-cursors/src/main/java/org/zalando/nakadi/service/CursorsService.java:237]:
```suggestion
                            .map(p -> p.toLastDeadLetterOffset(null))
```
I think that also resets the failed commits to 0?  Or should we have a function to do both things explicitly?

**Comment by fghibellini** [comment]
👍

File: acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java [hunk 3]
```diff
@@ -39,8 +40,7 @@
 import java.util.concurrent.atomic.AtomicReference;
 import java.util.stream.Collectors;
 
-import static com.jayway.restassured.RestAssured.given;
-import static com.jayway.restassured.RestAssured.when;
+import static com.jayway.restassured.RestAssured.*;
```
Comment (a1exsh) [inline acceptance-test/src/acceptance-test/java/org/zalando/nakadi/webservice/hila/HilaAT.java:43]: doesn't checkstyle prohibit this?

**Pull request merged**